### PR TITLE
Getting started

### PR DESCRIPTION
**Pull request by littlemauli**
We have implemented 3 different locations, with picture and info.
We have embedded a weather widget that also shows the 3 different locations.

### CODE REVIEW

File: ClientApp/src/Pages/GettingStarted/GettingStarted.scss [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+            align-self: center;
+            flex-direction: column;
+            width: 100%;
+            justify-content: center;
+            align-items:center;
+        } 
+
+        body {
+
+            // background-color: rgb(92, 111, 216);
```
Comment (MichaelWalker) [inline ClientApp/src/Pages/GettingStarted/GettingStarted.scss]: should this be deleted now?

File: ClientApp/src/Pages/GettingStarted/GettingStarted.scss [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+            padding-bottom: 50px;
+        }
+    
+    }
+
+
+@media only screen and (min-width: 1000px){   
+
+   body {
+//    background-color: rgb(92, 111, 216);
```
Comment (MichaelWalker) [inline ClientApp/src/Pages/GettingStarted/GettingStarted.scss]: same here

File: ClientApp/src/Pages/GettingStarted/GettingStarted.scss [hunk 3]
```diff
@@ -0,0 +1,79 @@
+
+@media only screen and (max-width: 1000px) {
+        .ImageAndText{
```
Comment (MichaelWalker) [inline ClientApp/src/Pages/GettingStarted/GettingStarted.scss]: by convention this should be `image-and-text` - would you mind changing it?

File: ClientApp/src/Pages/GettingStarted/GettingStarted.scss [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+            align-items:center;
+        } 
+
+        body {
+
+            // background-color: rgb(92, 111, 216);
+            text-align: center;
+            
+            }
+        h3{
```
Comment (MichaelWalker) [inline ClientApp/src/Pages/GettingStarted/GettingStarted.scss]: these styles for body and h3 look like they will affect the whole site - ie not just this one page.
Is that intentional?

File: ClientApp/src/Pages/GettingStarted/GettingStarted.tsx [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+    else if (selectedLocation === Locations.Scotland) {
+        x = latScotland;
+        y = LonScotland;
+        loc = "Gairloch, Scotland"
+    }
+    else if (selectedLocation === Locations.Australia) {
+        x = LatAustralia;
+        y = LonAustralia;
+        loc = "Queensland, Australia"
+    }
```
Comment (MichaelWalker) [inline ClientApp/src/Pages/GettingStarted/GettingStarted.tsx]: this works totally fine, but I think we could make it a little bit cleaner with a little refactor.
Let's pull this whole thing out into its own little function, and then our component can use the data that its given.

ie
```ts
interface LocationData {
    latitude: number;
    longitude: number;
    label: string;
}

const canadaLocationData = {
    latitude: 53.135509,
    longitude: -57.660435,
    label: "NewFoundland, Canada",
};

const scotlandLocationData = { /* stuff */ };
const austrailiaLocationData = { /* stuff */ };

function getLocationData(location: Locations): LocationData {
    switch (location) {
        case: Locations.Canada:
            return canadaLocationData;
        case: Locations.Scotland:
            return scotlandLocationData;
        case: Locations.Australia:
            return australiaLocationData;
    }
}
```

Then out component can just do

```ts
const locationData = getLocationData();

const { data, isLoading, errorMessage } = useOpenWeather({
    key: 'stuff',
    lat: locationData.latitude,
    lon: locationData.longitude,
    lang: 'en',
    unit: 'metric'
});
```

File: ClientApp/src/Pages/GettingStarted/GettingStarted.tsx [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+
+    const { data, isLoading, errorMessage } = useOpenWeather({
+        key: '43887186b07bb5deaad592918ae44878',
```
Comment (MichaelWalker) [inline ClientApp/src/Pages/GettingStarted/GettingStarted.tsx]: this looks like it might be a secret API key...
Are you sure its OK for it to be made public in GitHub?

(if not you might need to cancel this one and generate a new one)

File: ClientApp/src/Pages/GettingStarted/GettingStarted.tsx [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+
+    return (
+        <div>
+            <div className="getting_started">
+
+                <div className='top-page-break'></div>
+
+                <h2>Getting Started: Find the Perfect Whalespotting Location</h2>
+                <div className="places">
+                    <h3 /*data-testid={"ScotlandButton"}*/ className={selectedLocation === Locations.Scotland ? 'selected-icon' : 'unselected-icon'} onClick={() => setSelectedLocation(Locations.Scotland)}>Scotland</h3>
```
Comment (MichaelWalker) [inline ClientApp/src/Pages/GettingStarted/GettingStarted.tsx]: few more little bits of commented out code here. :)

File: ClientApp/src/Pages/GettingStarted/GettingStarted.tsx [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+        unit: 'metric', // values are (metric, standard, imperial)
+    });
+
+    return (
+        <div>
+            <div className="getting_started">
+
+                <div className='top-page-break'></div>
+
+                <h2>Getting Started: Find the Perfect Whalespotting Location</h2>
```
Comment (MichaelWalker) [inline ClientApp/src/Pages/GettingStarted/GettingStarted.tsx]: Should this be an h1? Looks like it might be the title of the page.

File: ClientApp/src/Pages/GettingStarted/GettingStarted.tsx [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+                <div className="places">
+                    <h3 /*data-testid={"ScotlandButton"}*/ className={selectedLocation === Locations.Scotland ? 'selected-icon' : 'unselected-icon'} onClick={() => setSelectedLocation(Locations.Scotland)}>Scotland</h3>
+                    <h3 /*data-testid={"CanadaButton"}*/ className={selectedLocation === Locations.Canada ? 'selected-icon' : 'unselected-icon'} onClick={() => setSelectedLocation(Locations.Canada)}>Canada</h3>
+                    <h3 /*data-testid={"AustraliaButton"}*/ className={selectedLocation === Locations.Australia ? 'selected-icon' : 'unselected-icon'} onClick={() => setSelectedLocation(Locations.Australia)}>Australia</h3>
```
Comment (MichaelWalker) [inline ClientApp/src/Pages/GettingStarted/GettingStarted.tsx]: all these `h3`s look like they are being used as buttons... (or perhaps some kind of form input?)
If so... why not just make them buttons?

That way:
- styling for if they are pressed is easier
- its better for accessibility
- the browser understands what's happening, so will automatically give it sensible hover states, keyboard interaction etc etc

File: ClientApp/src/Pages/GettingStarted/GettingStarted.tsx [hunk 10]
```diff
@@ -0,0 +1,104 @@
...
+                <div>
+                    {selectedLocation === Locations.Canada ? <div>
+                        <LocationSummary location={LocationData.Canada} />
+                    </div> : null}
+                </div>
+                <div>
+                    {selectedLocation === Locations.Australia ? <div>
+                        <LocationSummary location={LocationData.Australia} />
+                    </div> : null}
+                </div>
```
Comment (MichaelWalker) [inline ClientApp/src/Pages/GettingStarted/GettingStarted.tsx]: this can also benefit from the refactor that we made earlier! :)

```ts
<div><LocationSummary location={locationData}></div>
```

File: ClientApp/src/Pages/GettingStarted/LocationData.json [hunk 11]
```diff
@@ -0,0 +1,34 @@
+
+
+{
```
Comment (MichaelWalker) [inline ClientApp/src/Pages/GettingStarted/LocationData.json]: Could we make this a typescript file rather than a JSON one?
That way we get all the nice type safety :) 

ie just
```ts
interface LocationData {
    name: string;
    imageSrc: string;
    description: string;
    // ...
}

export const scotland: LocationData = {
    name: "Gairloch, Wester Ross",
    // ...
}

File: ClientApp/src/Pages/GettingStarted/LocationSummary.tsx [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+}
+
+function LocationSummary(props: LocationSummary): JSX.Element {
+
+    return (
+        <div>
+            <h2> {props.location.name}</h2>
+            <div className='location-line-break'></div>
+            <div className='ImageAndText'>
+                <img className='locationImage' src={props.location.imagesrc} width='400px' height='400px' alt='Image of Location' />
```
Comment (MichaelWalker) [inline ClientApp/src/Pages/GettingStarted/LocationSummary.tsx]: could we style this in CSS rather than using the width and height inline?

File: package.json [hunk 13]
```diff
@@ -8,5 +8,8 @@
   },
   "keywords": [],
   "author": "",
-  "license": "ISC"
+  "license": "ISC",
+  "dependencies": {
+    "react-open-weather": "^1.1.3"
+  }
 }
```
Comment (MichaelWalker) [inline package.json]: Could you undo this change? 
I don't think there is any need to add anything to this file.

**Review by MichaelWalker — CHANGES_REQUESTED** [review]
Nice Work! :) 

Just just suggested a few little refactors, and there are a couple of places where there are commented out lines of code that I think can just be deleted.

File: ClientApp/src/Pages/GettingStarted/GetLocationData.tsx [hunk 14]
```diff
@@ -0,0 +1,68 @@
...
+        case Locations.Canada:
+            return canada ;
+        case Locations.Australia:
+            return australia;
+
+    }
+
+}
+
+export{GetLocationData}
```
Comment (ChristianMcc) [inline ClientApp/src/Pages/GettingStarted/GetLocationData.tsx]: formatting nitpick but could we fix the spacing here and get rid of the extra new lines in the functions above please :)

File: ClientApp/src/Pages/GettingStarted/GettingStarted.scss [hunk 15]
```diff
@@ -0,0 +1,60 @@
+
+    .image-and-text {
```
Comment (ChristianMcc) [inline ClientApp/src/Pages/GettingStarted/GettingStarted.scss]: I think this is all indented by one, a shift + alt + f should fix this!

File: ClientApp/src/Pages/GettingStarted/GettingStarted.scss [hunk 16]
```diff
@@ -0,0 +1,60 @@
...
+        justify-content: center;
+        align-items: center;
+    }
+
+    .getting-started {
+        text-align: center;
+    }
+
+    .selected-icon {
+        padding-left: 5%;
```
Comment (ChristianMcc) [inline ClientApp/src/Pages/GettingStarted/GettingStarted.scss]: I think this would better as something like 
```
.icon { 
     padding-left: 5%;
      padding-right: 5%;
      font-size: x-large;
}
.selected-icon {
    background-color: orange;
}
```

**Review by ChristianMcc — COMMENTED** [review]
Looking good! It would be good to add some tests here similar to mars rover where we test clicking the different locations and making sure hte page updates :)

File: ClientApp/src/Pages/GettingStarted/GettingStarted.test.tsx [hunk 17]
```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import { GettingStarted } from './GettingStarted';
+import { fireEvent, render, screen,act } from '@testing-library/react';
+import ReactDOM from 'react-dom';
+import { MemoryRouter } from 'react-router-dom';
+// import 'jest-dom/extend-expect';
```
Comment (ChristianMcc) [inline ClientApp/src/Pages/GettingStarted/GettingStarted.test.tsx]: Can delete this line :)

File: ClientApp/src/Pages/GettingStarted/GettingStarted.test.tsx [hunk 18]
```diff
@@ -0,0 +1,44 @@
...
+      expect(australiaButton.className).toBe('selected-icon')
+
+})
+
+test('LocationSummary for Australia is shown when Australia is clicked', () => {
+    const { getByTestId }  = render(<GettingStarted />)
+    const australiaButton = getByTestId('AustraliaButton')
+    expect(australiaButton.className).toBe('unselected-icon')
+    fireEvent.click(australiaButton)
+    // expect(screen.getByTestId('description')).toHaveTextContent("Queensland’s whale hot spot is at Hervey Bay, around 300 kilometres north of Brisbane. This tranquil spot is protected by Fraser Island, affording humpback whales the opportune location to hang around a while, either to mate, breed or nurse. Each year in July and August, the arrival of the humpback whales is celebrated at the Hervey Bay Whale Festival, with concerts, street parades and displays.");
```
Comment (ChristianMcc) [inline ClientApp/src/Pages/GettingStarted/GettingStarted.test.tsx]: can delete this too if you're not using it

**Review by ChristianMcc — APPROVED** [review]
Looks good! A couple of bits of commented code to delete but once that's done feel free to merge :)

**Review by emilypatsko — APPROVED** [review]

**Review by RoshniCh — APPROVED** [review]
